model/src/w3srcemd.F90

```diff
@@ -1378,9 +1395,123 @@ SUBROUTINE W3SRCE ( srce_call, IT, JSEA, IX, IY, IMOD,          &
                WRITE(740+IAPROC,*) 'min/max/sum(VDBT)=', minval(VDBT), maxval(VDBT), sum(VDBT)
         END IF
 #endif
+         IF (srce_call .eq. srce_imp_pre) THEN
+#ifdef W3_PDLIB
+      IF (LSLOC) THEN
+        IF (IMEM == 1) THEN
+          SIDT  = PDLIB_SI(JSEA) * DTG 
+          DO IK = 1, NK
+            JAC = CLATSL/CG1(IK) 
+            DO ITH = 1, NTH
+              ISP = ITH + (IK-1)*NTH
+#endif
+#ifdef W3_PDLIB
+              VD(ISP) = MIN(0., VD(ISP))
+              IF (LNEWLIMITER) THEN
+                MAXDAC = MAX(DAM(ISP),DAM2(ISP))
+              ELSE
+                MAXDAC = DAM(ISP)
+              ENDIF
+              FAKS   = DTG / MAX ( 1. , (1.-DTG*VD(ISP)))
+              DVS    = VS(ISP) * FAKS
+              DVS    = SIGN(MIN(MAXDAC,ABS(DVS)),DVS)
+              PreVS  = DVS / FAKS
+              eVS    = PreVS / CG1(IK) * CLATSL
+              eVD    = MIN(0.,VD(ISP))
+              B_JAC(ISP,JSEA)                   = B_JAC(ISP,JSEA) + SIDT * (eVS - eVD*SPEC(ISP)*JAC)
+              ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) = ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) - SIDT * eVD
+#endif
 
+#ifdef W3_DB1
+                eVS = VSDB(ISP) * JAC
+                eVD = MIN(0.,VDDB(ISP))
+#endif
+
+#ifdef W3_DB2
+                eVS = VSDB(ISP) * JAC
+                eVD = MIN(0.,VDDB(ISP))
+#endif
+
+#ifdef W3_PDLIB
+              B_JAC(ISP,JSEA)                   = B_JAC(ISP,JSEA) + SIDT * eVS
+              ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) = ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) - SIDT * eVD
+#endif
+
+#ifdef W3_TR1
+                eVS = VSTR(ISP) * JAC 
+                eVD = VDTR(ISP)
+#endif
+
+#ifdef W3_PDLIB
+              B_JAC(ISP,JSEA)                   = B_JAC(ISP,JSEA) + SIDT * eVS
+              ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) = ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) - SIDT * eVD
+#endif
+                     !IF (IX == 1232) write(740+IAPROC,*) IX, ISP, B_JAC(ISP,JSEA), ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)), &
+                     !                         & VSLN(ISP), VSIN(ISP), VDIN(ISP), VSDS(ISP), VDDS(ISP), &
+                     !                         !& VSTR(ISP), VDTR(ISP), VSDB(ISP), VDDB(ISP), VSNL(ISP), VDNL(ISP), &
+                     !                         & SPEC(ISP), U10ABS, U10DIR, USTAR 
+                     
+!/VEG1              eVS = eVS + VSVG(ISP) * JAC 
+!/VEG1              eVD = evD + MIN(0.,VDVG(ISP))
+!/PDLIB             B_JAC(ISP,JSEA)                   = B_JAC(ISP,JSEA) + SIDT * (eVS - eVD*VA(ISP,JSEA)*JAC)
+!/PDLIB             ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) = ASPAR_JAC(ISP,PDLIB_I_DIAG(JSEA)) - SIDT * eVD
+#ifdef W3_PDLIB
+            END DO
+#endif
+
+#ifdef W3_PDLIB
+          END DO
+#endif
+!                 IF (IT .gt. 0 .and. JSEA .eq. 100) THEN
```

Comment:
remove commented out code. also not the above and below #endif and #ifdefs are not needed.

Reply:
done

Reply:
With the clean up I've been able to look at this again.  I'm assuming you can have DB1, DB2, or TR1 without PDLIB, and if you look at the code section above I'm a little worried that you could have code that you do not want if you have DB1, DB2, or TR1 without PDLIB.  Should the #end if on 1421 be deleted and then the whole section through 1448 be in the ifdef W3_PDLIB?

Reply:
Also I still see the commented out code...